src/sb/subscriber.rs

```diff
@@ -3,21 +3,144 @@
+        self.options.history_request = history_request;
+        self
+    }
+
+    pub fn node_name(mut self, node_name: String) -> Self {
+        self.options.node_name = node_name;
+        self
+    }
+
+    pub fn queue_full_policy(mut self, queue_full_policy: QueueFullPolicy) -> Self {
+        self.options.queue_full_policy = queue_full_policy;
+        self
+    }
+
+    pub fn requires_publisher_history_support(
+        mut self,
+        requires_publisher_history_support: bool,
+    ) -> Self {
+        self.options.requires_publisher_history_support = requires_publisher_history_support;
+        self
+    }
+
+    pub fn create(mut self) -> Result<(st::Subscriber<T>, SampleReceiverToken), IceoryxError> {
+        self.options.subscribe_on_create = true;
+        let ffi_sub = ffi::Subscriber::new(self.service, self.instance, self.event, &self.options)
+            .ok_or(IceoryxError::SubscriberCreationFailed)?;
+
+        let subscriber = st::Subscriber {
+            ffi_sub: ffi::SubscriberRc::new(ffi_sub),
+            phantom: PhantomData,
+        };
+
+        Ok((subscriber, SampleReceiverToken {}))
+    }
+
+    pub fn create_mt(mut self) -> Result<(mt::Subscriber<T>, SampleReceiverToken), IceoryxError> {
+        self.options.subscribe_on_create = true;
+        let ffi_sub = ffi::Subscriber::new(self.service, self.instance, self.event, &self.options)
+            .ok_or(IceoryxError::SubscriberCreationFailed)?;
+
+        let subscriber = mt::Subscriber {
+            ffi_sub: ffi::SubscriberArc::new(ffi_sub),
+            phantom: PhantomData,
+        };
+
+        Ok((subscriber, SampleReceiverToken {}))
+    }
+
+    pub fn create_without_subscribe(mut self) -> Result<InactiveSubscriber<T>, IceoryxError> {
+        self.options.subscribe_on_create = false;
+        let ffi_sub = ffi::Subscriber::new(self.service, self.instance, self.event, &self.options)
+            .ok_or(IceoryxError::SubscriberCreationFailed)?;
+
+        Ok(InactiveSubscriber {
+            ffi_sub,
+            phantom: PhantomData,
+        })
+    }
+}
+
+pub struct SampleReceiverToken {}
+
+pub struct InactiveSubscriber<T> {
+    ffi_sub: Box<ffi::Subscriber>,
+    phantom: PhantomData<T>,
+}
+
+impl<T> InactiveSubscriber<T> {
+    fn from_ffi(ffi_sub: Box<ffi::Subscriber>) -> Self {
+        Self {
+            ffi_sub,
+            phantom: PhantomData,
+        }
+    }
+
+    pub fn subscribe(self) -> (st::Subscriber<T>, SampleReceiverToken) {
+        self.ffi_sub.subscribe();
+        (
+            st::Subscriber::new_from_ffi(self.ffi_sub),
+            SampleReceiverToken {},
+        )
+    }
+
+    pub fn subscribe_mt(self) -> (mt::Subscriber<T>, SampleReceiverToken) {
+        self.ffi_sub.subscribe();
+        (
+            mt::Subscriber::new_from_ffi(self.ffi_sub),
+            SampleReceiverToken {},
+        )
+    }
+
+    pub fn subscription_state(&self) -> SubscribeState {
```

Comment:
Didn't you forget to add `get_service_description`?

Reply:
I explicitly did not add it since there is no `ServiceDescription` in the Rust bindings and I'd rather refactor this to a `ServiceIdentifier` with a single string in the future.